datalad/support/gitrepo.py

```diff
@@ -671,21 +674,174 @@ def git_remote_update(self, name='', verbose=False):
+        if remote is None:
+            if refspec is not None:
+                # conflicts with using tracking branch or fetch all remotes
+                # For now: Just fail.
+                # TODO: May be check whether it fits to tracking branch
+                raise ValueError("refspec specified without a remote. (%s)" %
+                                  refspec)
+            if all:
+                remotes_to_fetch = self.repo.remotes
+            else:
+                # No explicit remote to fetch.
+                # => get tracking branch:
+                tb = self.repo.active_branch.tracking_branch().name
+                if tb:
+                    parts = tb.split('/')
+                    assert len(parts) == 2
+                    remotes_to_fetch = [self.repo.remote(parts[0])]
+                    refspec = parts[1]
+                else:
+                    # No remote, no tracking branch
+                    # => fail
+                    raise ValueError("No remote specified to fetch from nor a "
+                                     "tracking branch is set up.")
+        else:
+            remotes_to_fetch = [self.repo.remote(remote)]
+
+        for rm in remotes_to_fetch:
+            fetch_url = \
+                rm.config_reader.get('fetchurl'
+                                     if rm.config_reader.has_option('fetchurl')
+                                     else 'url')
+            if fetch_url.startswith('ssh:'):
+                cnct = ssh_manager.get_connection(fetch_url)
+                cnct.open()
+                # TODO: with git <= 2.3 keep old mechanism:
+                #       with rm.repo.git.custom_environment(GIT_SSH="wrapper_script"):
+                with rm.repo.git.custom_environment(
+                        GIT_SSH_COMMAND="ssh -S %s" % cnct.ctrl_path):
+                    rm.fetch(refspec=refspec, progress=progress, **kwargs)  # TODO: progress +kwargs
+            else:
+                rm.fetch(refspec=refspec, progress=progress, **kwargs)  # TODO: progress +kwargs
+
+        # TODO: fetch returns a list of FetchInfo instances. Make use of it.
+
+    def pull(self, remote=None, refspec=None, progress=None, **kwargs):
+        """
+        """
+        if remote is None:
+            if refspec is not None:
+                # conflicts with using tracking branch or fetch all remotes
+                # For now: Just fail.
+                # TODO: May be check whether it fits to tracking branch
+                raise ValueError("refspec specified without a remote. (%s)" %
+                                  refspec)
+            # No explicit remote to pull from.
+            # => get tracking branch:
+            tb = self.repo.active_branch.tracking_branch().name
+            if tb:
+                parts = tb.split('/')
+                assert len(parts) == 2
+                remote = self.repo.remote(parts[0])
+                refspec = parts[1]
```

Comment:
before even "Centralized" worth extracting this duplicated logic into a helper function (`split_tracking_branch`?) so it could also be unit-tested as well